layouts/partials/header.html

```diff
@@ -1,16 +1,28 @@
 <nav class="navigation">
   <section class="container">
-    <a class="navigation-title" href="{{ .Site.BaseURL }}">
+    <a class="navigation-title" href="{{ .Site.Language }}">
       {{ .Site.Title }}
     </a>
-    {{ with .Site.Menus.main }}
+    {{ if .Site.Menus.main }}
```

Comment:
this whole block can be done as follow, IMO it's cleaner and without the repetition of `language-selector.html`:

```
<ul class="nav navbar-nav {{ if $.Site.Params.rtl }} navbar-left {{ else }} navbar-right {{ end }}">
  {{ with .Site.Menus.main }}
    {{ range sort . }}
      <li>
        <a href="{{ .URL | absLangURL }}">{{ i18n .Name }}</a>
      </li>
    {{ end }}
  {{ end }}

  {{ if .Site.IsMultiLingual }}
    {{ $node := . }}
    {{ .Scratch.Set "separator" true }}

    {{ range .Site.Home.AllTranslations }}
      {{ $language := printf "%s" .Language }}
      {{ if ne $.Site.Params.Langcode $language }}
        {{ if $node.Scratch.Get "separator" }}
          <li>
            <span>|</span>
          </li>
          {{ $node.Scratch.Set "separator" false }}
        {{ end }}
        <li>
          <a href="{{ .Permalink }}">{{ .Language.LanguageName }}</a>
        </li>
      {{ end }}
    {{ end }}
  {{ end }}
</ul>
```
I personally don't like the separator, but I added it here as an example (and it would be cool to make that configurable).